### PR TITLE
PAAS-1736: add recursive mode to vaultSecretsList action

### PR DESCRIPTION
**Pull request by adoux**
JIRA issue: https://jira.jahia.org/browse/PAAS-1736

Short description:

### CODE REVIEW

File: mixins/common.yml [hunk 1]
```diff
@@ -411,23 +411,57 @@ actions:
...
+              # else we just print the subpath with the trailing slash
+              if [ "$recursive" = "true" ]; then
+                # We remove the trailing slash from the full path
+                list_secrets_to_file $(echo -n $full_path | head -c -1)
+              else
+                echo $full_path >> $final_secrets_list
+              fi
+            else
+              echo $full_path >> $final_secrets_list
+            fi
```
Comment (Ronan-Gorain) [inline mixins/common.yml:456]:
```suggestion
            # If recursive and "secret" ends with a slash, means it's a subpath
            #     We remove the trailing slash from the full path
            # else we just print the subpath with the trailing slash
            if [ "$recursive" = "true" ] && echo $path_end | grep -qE "/$"; then
                list_secrets_to_file $(echo -n $full_path | head -c -1)
            else
              echo $full_path >> $final_secrets_list
            fi
```

Comment (Ronan-Gorain) [reply mixins/common.yml:456]: It's just a suggestion (tested). If you disagree, I'll merge.

Comment (Ronan-Gorain) [reply mixins/common.yml:456]: I'm also not sure why the comment talks about removing the trailing slash from the full path as I don't see this operation on lines arround

Comment (adoux) [reply mixins/common.yml]: it's the line right after (`head -c -1`) :-)

**Review by Ronan-Gorain — APPROVED** [review]

**Pull request merged**